### PR TITLE
Fix surface-shader

### PR DESCRIPTION
**Pull request by GengineJS**

### CODE REVIEW

File: cocos/rendering/custom/web-pipeline-types.ts [hunk 1]
```diff
@@ -901,37 +901,48 @@ export class RenderDrawQueue {
...
-        while (!res.done) {
-            res.value.clear();
-            res = it.next();
-        }
-        this.queue.clear();
+        this.passInstances.clear();
+        const instanceBuffers = this.instanceBuffers;
+        instanceBuffers.forEach((instance) => {
+            instance.clear();
+        });
```
Comment (minggo) [inline cocos/rendering/custom/web-pipeline-types.ts:935]: Should also clear instanceBuffers?

**Review by star-e — APPROVED** [review]

File: cocos/rendering/custom/web-pipeline-types.ts (hunk 1 above)
Comment (star-e) [reply cocos/rendering/custom/web-pipeline-types.ts:935]: I think gfx instance buffers are reused. They are not recreated every frame.

Comment (GengineJS) [reply cocos/rendering/custom/web-pipeline-types.ts:935]: The instanceBuffers here will be reused, so there's no need to clear them

Comment (minggo) [reply cocos/rendering/custom/web-pipeline-types.ts:935]: And the size of it is not changed?

Comment (GengineJS) [reply cocos/rendering/custom/web-pipeline-types.ts:935]: The reuse logic is here： image

**Review by minggo — APPROVED** [review]

**Pull request merged**